test.js

```diff
@@ -416,6 +416,203 @@ Object.entries(POLYFILLS).forEach(([name, module]) => {
+              'Unsafe legacy lifecycles will not be called for components using new component APIs.\n\n' +
+                'ComponentWithWillReceiveProps uses getSnapshotBeforeUpdate() but also contains the following legacy lifecycles:\n' +
+                '  componentWillReceiveProps\n\n' +
+                'The above lifecycles should be removed. Learn more about this warning here:\n' +
+                'https://fb.me/react-async-component-lifecycle-hooks'
+            );
+
+            console.error.mockClear();
+            class ComponentWithUnsafeWillMount extends React.Component {
+              UNSAFE_componentWillMount() {}
+              componentDidUpdate() {}
+              getSnapshotBeforeUpdate() {}
+              render() {
+                return null;
+              }
+            }
+
+            expect(console.error.mock.calls).toHaveLength(0);
+            polyfill(ComponentWithUnsafeWillMount);
+            expect(console.error.mock.calls).toHaveLength(1);
+            expect(console.error.mock.calls[0][0]).toEqual(
+              'Unsafe legacy lifecycles will not be called for components using new component APIs.\n\n' +
+                'ComponentWithUnsafeWillMount uses getSnapshotBeforeUpdate() but also contains the following legacy lifecycles:\n' +
+                '  UNSAFE_componentWillMount\n\n' +
+                'The above lifecycles should be removed. Learn more about this warning here:\n' +
+                'https://fb.me/react-async-component-lifecycle-hooks'
+            );
+
+            console.error.mockClear();
+
+            class ComponentWithUnsafeWillReceiveProps extends React.Component {
+              UNSAFE_componentWillReceiveProps() {}
+              componentDidUpdate() {}
+              getSnapshotBeforeUpdate() {}
+              render() {
+                return null;
+              }
+            }
+
+            expect(console.error.mock.calls).toHaveLength(0);
+            polyfill(ComponentWithUnsafeWillReceiveProps);
+            expect(console.error.mock.calls).toHaveLength(1);
+            expect(console.error.mock.calls[0][0]).toEqual(
+              'Unsafe legacy lifecycles will not be called for components using new component APIs.\n\n' +
+                'ComponentWithUnsafeWillReceiveProps uses getSnapshotBeforeUpdate() but also contains the following legacy lifecycles:\n' +
+                '  UNSAFE_componentWillReceiveProps\n\n' +
+                'The above lifecycles should be removed. Learn more about this warning here:\n' +
+                'https://fb.me/react-async-component-lifecycle-hooks'
+            );
+
+            console.error.mockClear();
+
+            class ComponentWithUnsafeWillUpdate extends React.Component {
+              UNSAFE_componentWillUpdate() {}
+              componentDidUpdate() {}
+              getSnapshotBeforeUpdate() {}
+              render() {
+                return null;
+              }
+            }
```

Comment:
My heart says to DRY this code up, but I think the repetition is actually more readable than trying to move this into a helper method.

Reply:
Yeah ☹️ It is very verbose though, agreed.